### DISCUSSION TITLE
How to get the axis of a revolute joint?

### DISCUSSION THREAD

**Discussion by 2352878630** [stack-of-tasks/pinocchio#2196] [Q&A] [answered]
I want to get a `Vector3` type variable that represents the axis of a revolute joint.
For example, in URDF file, I already have a joint with `<axis xyz="1 0 0"/>`. Now I want to get a variable with a value of `1 0 0` and a type of `Vector3`

Thank you very much !

**Comment by jcarpent** [comment, marked as answer]
Unfortunately, there is not yet a direct helper to extract it.
Yet, you can recover this information partially as follows:
```python
import pinocchio as pin
joint_model = pin.JointModelRX()
joint_data = joint_model.createData()
axis = joint_data.S[3:]
```

@jorisv Will you have time to add the `getMotionAxis` helper to all revolute and prismatic joints in Pinocchio and expose it to Python?

**Reply by jorisv** [reply]
I will check this. We also plan to use the new boost::variant eigenpy bindings to better expose joint in Python.

**Reply by 2352878630** [reply]
Thank you very much!

**Reply by joao-pm-santos96** [reply]
Hi @jorisv ! Using your snippet, I get `AttributeError: 'JointModelRX' object has no attribute 'createData'`. Im on pin version 2.7.0. Can you help me please?

**Reply by jorisv** [reply]
Hi @joao-pm-santos96,

Sorry this code snippet is not working in Pinocchio 2. It's only working in Pinocchio 3 that should be released at the end of the month.
I don't see a way to retrieve the S content in Pinocchio 2 (the S type is not binded).

You can get a Pinocchio 3 preview version [by following this instructions](https://github.com/agimus-project/winter-school-2023?tab=readme-ov-file#conda-installation-mac-os-intel-mac-os-arm-linux-x64)

**Reply by joao-pm-santos96** [reply]
Good news! I am exited to see the amazing thing pinocchio3 will bring! Keep up the amazing work! Thank you! 😃
